krod/core/engine.py

```
@@ -159,6 +193,63 @@ def _load_modules(self) -> Dict[str, Any]:
+        Returns:
+            Response dictionary
+        """
+        try:
+            # Get or create conversation
+            conversation = self.get_conversation(user_id, session_id)
+            
+            # Add user message to conversation
+            conversation.add_message(query, "user", **kwargs)
+            
+            # Process the query (your existing logic here)
+            response = self._process_query(query)
```

Pylint (3.3.7)

[error] Instance of 'KrodEngine' has no '_process_query' member; maybe 'process_query'?

(E1101)